### PR TITLE
Batching idx

### PR DESCRIPTION
**Pull request by YifanC**
make the batching work at track level by padding the data

### CODE REVIEW

File: optimize/dataio.py [hunk 1]
```diff
@@ -30,21 +30,33 @@ def __init__(self, filename, swap_xz=True):
             tracks['z_end'] = x_end
             tracks['z'] = x
 
+        tracks = tracks[200:250]
```
Comment (sgasioro) [inline optimize/dataio.py:33]: This should probably be removed

File: optimize/dataio.py [hunk 2]
```diff
@@ -30,21 +30,33 @@ def __init__(self, filename, swap_xz=True):
...
+        self.index = []
+        num_tracks = 0
+        all_tracks = []
+        all_events = np.unique(tracks['eventID'])
+        for ev in all_events:
+            track_set = np.unique(tracks[tracks['eventID'] == ev]['trackID'])
+            for trk in track_set:
+                # basic track selection
+                trk_msk = (tracks['eventID'] == ev) & (tracks['trackID'] == trk)
+                if max(tracks[trk_msk]['z']) - min(tracks[trk_msk]['z']) > 3:
```
Comment (sgasioro) [inline optimize/dataio.py:47]: Should we make this cut value configurable?

**Comment by sgasioro** [comment]
Other than those two comments, I think this looks good!

File: optimize/dataio.py (hunk 1 above)
Comment (YifanC) [reply optimize/dataio.py]: see commit 59b987ea1671cb42e86c9f929c059f27ffdf8ccf

File: optimize/dataio.py (hunk 2 above)
Comment (YifanC) [reply optimize/dataio.py]: see commit 6ac3d72d8a4caf62ff1a5dd1d5d31353a7e57938

**Pull request merged**